sympy/sets/handlers/issubset.py

```diff
@@ -35,17 +35,15 @@ def is_subset_sets(a_interval, b_fs): # noqa:F811
 def is_subset_sets(a_interval, b_u): # noqa:F811
     if all(isinstance(s, (Interval, FiniteSet)) for s in b_u.args):
         intervals = [s for s in b_u.args if isinstance(s, Interval)]
-        if all(fuzzy_bool(a_interval.start < s.start) for s in intervals):
-            return False
-        if all(fuzzy_bool(a_interval.end > s.end) for s in intervals):
+        sol = S.EmptySet
+        for interval in intervals:
+            sol = sol.union(a_interval.intersect(interval))
```

Comment:
We shouldn't depend on the evaluation code to resolve a query like this. This should not create new unions or intersections.

Reply:
Okay, I get what you are saying. The other dispatchers rely on simple comparison for their working . The problem here is you are not comparing just one interval to another, you are comparing 1 interval to 'n' intervals. 
Would it be okay, to use a temporary interval to maintain a 'check' , as to how much of the interval has been 'covered '?  This would be similar to having intersection of intervals but without using the ``intersect`` operation.

Reply:
Do you have any ideas that might work? I can only think of taking the intersection and then comparing it with the final interval. 
 Moreover I think this is quite erroneous 
https://github.com/sympy/sympy/blob/9107b9d7ec4449b888cfd41a540008c511b5a260/sympy/sets/handlers/issubset.py#L42-L48
As in #21107 , this check will always fail, if ``a_interval`` is ``Interval(-oo,oo)`` or any Interval which extends like ``Infinity`` (regardless of sign) and thus would always return ``None ``. 
```
>>> a = Interval(-oo, 2)
>>> b = Union(Interval(-oo,-20), Interval(4,10))
>>> print(a.is_subset(b))
None
``` 
This can be fixed by doing  `` not a_interval.measure.is_zero `` , but it too gives ``None`` though

Reply:
> Moreover I think this is quite erroneous

We need to be clear that in three-valued logic it is always valid to return `None` because that means "don't know" or "maybe" which is always a valid response. Although it might sometimes be better to return True or False some cases are undecidable so there always needs to be the possibility of returning None.

We then need to decide how hard we want to work (in terms of computation time and lines of code) in order to get True or False more often. The implementation you are pointing at here is intended to be something that might work in some fairly generic cases but is expected to be fast rather than complete.

If you can think of a common class of problems that is not well handled currently and write some fast routine for handling that then that can be added. However you need to bear in mind that the more things we check the slower `is_subset` will be.

If the particular reason you want to add this to `is_subset` is to simplify some kind of intersection or something then I think it would be better for us to add more code in simplification rather than in evaluation.

> This can be fixed by doing `not a_interval.measure.is_zero` , but it too gives `None` though

Since `is_zero` is a three-valued bool you need to use `fuzzy_not` rather than `not` in order to correctly handle the `None` case.

Reply:
I cannot comment on how fast/slow this would be, but I can attest that it is *a* right way to go about this. I don't think terming it generic is right.


> Since `is_zero` is a three-valued bool you need to use `fuzzy_not` rather than `not` in order to correctly handle the `None` case.

Thanks for this insight,  but what I actually meant was, `` print(Interval(-oo, oo).measure.is_nonzero) `` Prints False on the console. 
So either we can use `` fuzzy_not`` as you mentioned or a case can be added in that dispatcher to check if  if the interval is ``Reals`` .  
For this case( the case where the Interval is Reals) is easy to check  like 
``` 
intervals = [s for s in b_u.args if isinstance(s, Interval)]
if len(intervals) == 1:
 inter  = intervals[0] 
  if inter.start == -oo and inter.end == oo:
   return True  
 else:
  return False 
else:
 return False
```
So, Should a case for when  `` interval `` is ``Reals`` be added?

Reply:
You need to use `is_extended_nonzero` to handle the infinite case correctly.

Reply:
> So, Should a case for when `interval` is `Reals` be added?

You could add `if a_interval is S.Reals and any(s is S.Reals for s in intervals): return True` to handle the case of Reals being in an unevaluated Union.